odilo.py

```diff
@@ -0,0 +1,581 @@
+            return None
+
+    def get_availability(self, record_id):
+        url = self.RECORD_AVAILABILITY_ENDPOINT.replace('{recordId}', record_id)
+        status_code, headers, content = self.get(url)
+        content = json.loads(content)
+
+        if status_code == 200 and len(content) > 0:
+            return content
+        else:
+            self.log.warn('Cannot retrieve availability for record: ' + record_id + ' response http ' + status_code)
+            if content:
+                self.log.warn(content)
+            return None
+
+    @staticmethod
+    def _do_get(url, headers, **kwargs):
+        # More time please
+        if not kwargs:
+            kwargs = {"timeout": 60}
+        else:
+            kwargs['timeout'] = 60
+
+        if 'allow_redirects' not in kwargs:
+            kwargs['allow_redirects'] = True
+
+        response = HTTP.get_with_timeout(url, headers=headers, **kwargs)
+        return response.status_code, response.headers, response.content
+
+    @staticmethod
+    def _do_post(url, payload, headers, **kwargs):
+        # More time please
+        if not kwargs:
+            kwargs = {"timeout": 60}
+        else:
+            kwargs['timeout'] = 60
+
+        return HTTP.post_with_timeout(url, payload, headers=headers, **kwargs)
+
+
+class MockOdiloAPI(OdiloAPI):
+    @classmethod
+    def mock_collection(cls, _db):
+        library = DatabaseTest.make_default_library(_db)
+        collection, ignore = get_one_or_create(
+            _db, Collection,
+            name="Test Odilo Collection",
+            create_method_kwargs=dict(
+                external_account_id=u'library_id_123',
+            )
+        )
+        integration = collection.create_external_integration(
+            protocol=ExternalIntegration.ODILO
+        )
+        integration.password = u'abcdef123hijklm'
+        library.collections.append(collection)
+        return collection
+
+
+class OdiloRepresentationExtractor(object):
+    """Extract useful information from Odilo's JSON representations."""
+
+    log = logging.getLogger("OdiloRepresentationExtractor")
+
+    format_data_for_odilo_format = {
+        "ACSM": (
+            Representation.PDF_MEDIA_TYPE, DeliveryMechanism.ADOBE_DRM
+        ),
+        "PDF": (
+            Representation.PDF_MEDIA_TYPE, DeliveryMechanism.STREAMING_TEXT_CONTENT_TYPE
+        ),
+        "EPUB": (
+            Representation.EPUB_MEDIA_TYPE, DeliveryMechanism.STREAMING_TEXT_CONTENT_TYPE
+        ),
+        "MP3": (
+            Representation.MP3_MEDIA_TYPE, DeliveryMechanism.STREAMING_AUDIO_CONTENT_TYPE
+        ),
+        "MP4": (
+            Representation.MP4_MEDIA_TYPE, DeliveryMechanism.STREAMING_VIDEO_CONTENT_TYPE
+        ),
+        "WMV": (
+            Representation.WMV_MEDIA_TYPE, DeliveryMechanism.STREAMING_VIDEO_CONTENT_TYPE
+        ),
+        "JPG": (
+            Representation.JPEG_MEDIA_TYPE, DeliveryMechanism.NO_DRM
+        ),
```

Comment:
Again, I'm not sure what JPEGs you're delivering. I admit it's theoretically possible but I'd like to understand more so I can be sure about my recommendations.

Reply:
We deliver the images as a whole as I explained in a previous comment.

Reply:
OK, there's nothing wrong with this, it just seems a little weird to me, so go for it.